2004-05-14  Jakub Jelinek  <jakub@redhat.com>

	* gcc/config/ia64/ia64.h (TARGET_SWITCHES): Add -m64.
	* gcc/config/ia64/t-glibc (MULTILIB_OPTIONS, MULTILIB_DIRNAMES,
	MULTILIB_EXCEPTIONS, MULTILIB_OSDIRNAMES): Set.
	* gcc/config/ia64/linux.h (MULTILIB_DEFAULTS): Define.

--- gcc/config/ia64/ia64.h.jj	2004-03-12 08:14:28.000000000 -0500
+++ gcc/config/ia64/ia64.h	2004-05-14 07:56:35.000000000 -0400
@@ -168,6 +168,8 @@ extern int ia64_tls_size;
 
 #define TARGET_SWITCHES							\
 {									\
+  { "64",		0,						\
+      N_("Generate 64-bit code") },					\
   { "big-endian",	MASK_BIG_ENDIAN,				\
       N_("Generate big endian code") },					\
   { "little-endian",	-MASK_BIG_ENDIAN,				\
--- gcc/config/ia64/t-glibc.jj	2003-03-26 10:56:04.000000000 -0500
+++ gcc/config/ia64/t-glibc	2004-05-14 08:30:15.000000000 -0400
@@ -1 +1,6 @@
 LIB2ADDEH += $(srcdir)/config/ia64/fde-glibc.c
+
+MULTILIB_OPTIONS = m64/m32
+MULTILIB_DIRNAMES = 64 32
+MULTILIB_EXCEPTIONS = m32
+MULTILIB_OSDIRNAMES = ../lib64 ../lib
--- gcc/config/ia64/linux.h.jj	2004-02-10 07:20:51.000000000 -0500
+++ gcc/config/ia64/linux.h	2004-05-14 07:21:27.000000000 -0400
@@ -55,6 +55,8 @@ do {						\
 #undef LINK_EH_SPEC
 #define LINK_EH_SPEC ""
 
+#define MULTILIB_DEFAULTS { "m64" }
+
 /* Do code reading to identify a signal frame, and set the frame
    state data appropriately.  See unwind-dw2.c for the structs.  */
 
